packages/@aws-cdk/aws-ecs/lib/cluster.ts

```diff
@@ -267,9 +280,15 @@ export class EcsOptimizedAmi implements ec2.IMachineImage {
     if (props && props.generation) {      // generation defined in the props object
       if (props.generation === ec2.AmazonLinuxGeneration.AMAZON_LINUX && this.hwType !== AmiHardwareType.STANDARD) {
         throw new Error(`Amazon Linux does not support special hardware type. Use Amazon Linux 2 instead`);
+      } else if (props.windowsVersion) {
+        throw new Error('"windowsVersion" and Linux image "generation" cannot be both set');
       } else {
         this.generation = props.generation;
       }
+    } else if (props && props.windowsVersion) {
+      if (this.hwType !== AmiHardwareType.STANDARD) {
+        throw new Error('Windows Server does not support special hardware type. Use Amazon Linux 2 instead');
```

Comment:
Remove "use Amazon Linux 2 instead". I am assuming that if a user explicitly specified they wanted Windows, they should probably just not specify the HW type and not use Linux instead... :-)

Reply:
Done